src/registry/mod.rs

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Commands to manage a semantic convention registry.
+
+use clap::{Args, Subcommand};
+
+use crate::registry::generate::GenerateRegistry;
+use crate::registry::resolve::ResolveRegistry;
+use crate::registry::search::SearchRegistry;
+use check::CheckRegistry;
+use weaver_cache::Cache;
+use weaver_logger::Logger;
+
+use crate::registry::stats::StatsRegistry;
+
+mod check;
+mod generate;
+mod resolve;
+mod search;
+mod stats;
+
+/// Parameters for the `registry` command
+#[derive(Debug, Args)]
+pub struct RegistryCommand {
+    /// Define the sub-commands for the `registry` command
+    #[clap(subcommand)]
```

Comment:
Haven't interacted with Clap much, but this is really nice.

Reply:
Yes Clap is really nice. There is also a support various shell completions and few others things that we need to enable.